Index: src/lib/Libifl/tm.c
===================================================================
--- src/lib/Libifl/tm.c	(revision 6091)
+++ src/lib/Libifl/tm.c	(working copy)
@@ -1739,8 +1739,7 @@
     return TM_ENOTCONNECTED;
 
   /* write the pid so the adopted process can be part of the cpuset if needed */
-
-  if (diswsi(local_conn,sid) != DIS_SUCCESS)
+  if (diswsi(local_conn, pid) != DIS_SUCCESS)
     return TM_ENOTCONNECTED;
 
   /* send job or alternative id */
