diff --git a/src/Makefile.template b/src/Makefile.template
index cd186f1..543c98f 100644
--- a/src/Makefile.template
+++ b/src/Makefile.template
@@ -196,7 +196,6 @@ install-libshogun: $(TEMPLATE_TARGET)
 	rm -f $(DESTDIR)$(LIBDIR)/libshogun.$(EXT_LIB) $(DESTDIR)$(LIBDIR)/$(TEMPLATE_SONAME)
 	cd $(DESTDIR)$(LIBDIR) && ( ln -s $(TEMPLATE_TARGET) libshogun.$(EXT_LIB); \
 	ln -s $(TEMPLATE_TARGET) $(TEMPLATE_SONAME) )
-	-ldconfig
 
 install-python_static: $(TEMPLATE_TARGET)
 	install -d -m755 $(DESTDIR)$(PYDIR)
