--- a/src/shogun/mathematics/arpack.cpp	2011-08-31 19:06:16.000000000 -0400
+++ b/src/shogun/mathematics/arpack.cpp	2011-11-03 12:19:05.863397865 -0400
@@ -139,9 +139,9 @@
 			{
 				// compute (workd+ipntr[1]-1) = A*(workd+ipntr[0]-1)
 				cblas_dsymv(CblasColMajor,CblasUpper,
-				            n,1.0,matrix,n,
-				            (workd+ipntr[0]-1),1,
-				            0.0,(workd+ipntr[1]-1),1);
+				            (const int) n, (const double) 1.0, (const double*) matrix,
+				            (const int) n, (const double*) (workd+ipntr[0]-1), (const int) 1,
+				            (const double) 0.0, (double*) (workd+ipntr[1]-1), (const int) 1);
 			}
 			if (mode==3)
 			{
