--- numpy/distutils/system_info.py.bak	2013-10-22 17:56:47.658413792 -0700
+++ numpy/distutils/system_info.py	2013-10-22 17:31:27.597200602 -0700
@@ -1138,9 +1138,15 @@
         lib_dirs = self.get_lib_dirs()
 
         lapack_libs = self.get_libs('lapack_libs', self._lib_names)
-        info = self.check_libs(lib_dirs, lapack_libs, [])
-        if info is None:
-            return
+
+        for d in lib_dirs:
+            lapack = self.check_libs(d,lapack_libs,[])
+            if lapack is not None:
+                info = lapack
+                break
+        else:
+            return
+
         info['language'] = 'f77'
         self.set_info(**info)
 
@@ -1515,9 +1521,15 @@
         lib_dirs = self.get_lib_dirs()
 
         blas_libs = self.get_libs('blas_libs', self._lib_names)
-        info = self.check_libs(lib_dirs, blas_libs, [])
-        if info is None:
+
+        for d in lib_dirs:
+            blas = self.check_libs(d,blas_libs,[])
+            if blas is not None:
+                info = blas
+                break
+        else:
             return
+
         info['language'] = 'f77'  # XXX: is it generally true?
         self.set_info(**info)
 
